runtime/common/KernelWrapper.h

```diff
@@ -0,0 +1,347 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#pragma once
+#include "cudaq/platform.h"
+#include "cudaq/utils/registry.h"
+#include <cstdint>
+#include <cstring>
+#include <string>
+#include <vector>
+
+/*! \file KernelWrapper.h
+    \brief Utility classes to support library-mode kernel launch.
+
+    This header file defines classes supporting launching quantum kernels in
+   library mode (without the bridge) when using the remote platform.
+    Specifically, it provides a layer of indirection, `cudaq::invokeKernel`,
+   similar to the `altLaunchKernel` function when using the bridge. In addition,
+   argument packing (for serialization) is handled by C++ templates in lieu of
+   bridge-generated functions. These utilities will only be used for the remote
+   platform target in library-mode (guarded by pre-defined macros).
+*/
+
+namespace cudaq {
+
+/// @brief Wrapper for kernel address
+///
+/// This automatically handles kernels defined as struct/class or free
+/// functions.
+class KernelAddress {
+public:
+  KernelAddress() = default;
+  explicit KernelAddress(uint64_t address) : m_addr(address) {}
+
+  template <typename T,
+            typename = std::enable_if_t<!std::is_class_v<std::decay_t<T>>>>
+  static KernelAddress fromKernel(T *ptr) {
+    return KernelAddress(
+        static_cast<uint64_t>(reinterpret_cast<uintptr_t>(ptr)));
+  }
+
+  template <typename T,
+            typename = std::enable_if_t<std::is_class_v<std::decay_t<T>>>>
+  static KernelAddress fromKernel(T &&ref) {
+    return KernelAddress(
+        static_cast<uint64_t>(reinterpret_cast<uintptr_t>(&ref)));
+  }
+
+  template <typename T>
+  std::enable_if_t<std::is_pointer<T>::value, T> toPtr() const {
+    return reinterpret_cast<T>(static_cast<uintptr_t>(m_addr));
+  }
+
+private:
+  uint64_t m_addr = 0;
```

Comment:
Q: is the `m_` prefix a (informal) coding convention in the runtime?

Reply:
I'm not aware of any stated coding convention in this regard. Personally, I often use `m_` prefix to denote private member variables.